packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -134,55 +137,76 @@ export const formatPriceEvents = async (
         ancillaryData: event.args.ancillaryData,
         proposedPrice: ethers.utils.formatEther(event.args.proposedPrice),
         eventIndex: event.logIndex,
+        oracleAddress: event.address,
       };
     })
   );
 };
 
 export const getPolymarketMarkets = async (params: MonitoringParams): Promise<PolymarketMarket[]> => {
+  const markets = [];
+  const pagination = 100;
+  let offset = 0;
+
   const whereClause =
     "uma_resolution_status!='settled'" +
     " AND uma_resolution_status!='resolved'" +
     " AND question_ID IS NOT NULL" +
     " AND clob_Token_Ids IS NOT NULL";
 
-  const query = `
-    {
-      markets(where: "${whereClause}") {
-        resolvedBy
-        questionID
-        createdAt
-        question
-        outcomes
-        outcomePrices
-        liquidityNum
-        volumeNum
-        clobTokenIds
-        endDate
-        umaResolutionStatus
+  let moreMarketsAvailable = true;
+  while (moreMarketsAvailable) {
+    const query = `
+      {
+        markets(where: "${whereClause}", limit: ${pagination}, offset: ${offset}) {
+          resolvedBy
+          questionID
+          createdAt
+          question
+          outcomes
+          outcomePrices
+          liquidityNum
+          volumeNum
+          clobTokenIds
+          endDate
+          umaResolutionStatus
+        }
       }
+    `;
+
+    const { markets: polymarketContracts } = (await request(params.graphqlEndpoint, query)) as any;
+
+    // Remove markets with 1 week old endDate or more. So we only monitor markets that ended in the last week
+    // (or are still ongoing).
+    const now = Math.floor(Date.now() / 1000);
+    const oneWeek = 60 * 60 * 24 * 7;
+
+    const filtered = polymarketContracts.filter((contract: { [k: string]: any }) => {
+      const parsedDateTime = new Date(contract.endDate).getTime();
+      if (isNaN(parsedDateTime)) return true; // If endDate is not a valid date we keep the market.
```

Comment:
Just to double check: this is expected/common, right? Just want to make sure we're not ignoring errors or anything.